### DISCUSSION TITLE
What is the best practice of poetry using different python version?

### DISCUSSION THREAD

**Discussion by iamcodingcat** [python-poetry/poetry#8714] [Q&A]
I am newbie to poetry and I explored it. I have a question when using poetry with different python version. According to official docs, poetry is default using `virtualenv` tools for creating python virtual environment. So, I want to use python 3.8 in poetry, and I revise `toml` file from python 3.12 to python 3.8. But I encountered below error.
```bash
The currently activated Python version 3.12.0 is not supported by the project (3.9).
Trying to find and use a compatible version.

Poetry was unable to find a compatible version. If you have one, you can explicitly use it via the "env use" command.
```
After the error, I searched more in [docs](https://python-poetry.org/docs/managing-environments/) about managing environments. At the docs, it recommends `pyenv` tools for using other python downgraded version, but I see that pyenv don't supported well.. 

So, I wonder that the best practice of poetry of using different python version. What the best tool of managing virtual environment? Of course, we should not use downgraded python version.. I wonder it.

**Comment by radoering** [comment]
You can just [switch between environments](https://python-poetry.org/docs/managing-environments/#switching-between-environments).

**Reply by iamcodingcat** [reply]
Thanks. I will use pyenv for changing python version.
